easypy/concurrency.py

```diff
@@ -313,19 +317,36 @@ def add_details(exc):
 
         add_details(self)
 
-        for exc in self.actual:
-            with buff.indent("{.__class__.__qualname__}", exc):
+        should_coalesce_duplicated_tbs = False if (
+                os.getenv('EASYPY_MULTIEXCEPTION_COALESCE_DUPLICATED_TBS','') in ['false']) else True
+        for exc_num, exc in enumerate(self.actual, 1):
+            show_traceback = hasattr(exc, "__traceback__") and getattr(exc, 'traceback', None) is not False
+            full_exc_num_with_same_tb = None
+            full_exc_num = f"{exc_num_prefix}{exc_num}"
+
+            if should_coalesce_duplicated_tbs and show_traceback:
+                if tb := tuple((frame.filename, frame.lineno) for frame in extract_tb(exc.__traceback__)):
+                    if tb in unique_tbs_to_exc_num:
+                        full_exc_num_with_same_tb = unique_tbs_to_exc_num[tb]
+                    else:
+                        unique_tbs_to_exc_num[tb] = full_exc_num
+
+
+            with buff.indent(f"{{.__class__.__qualname__}} ({full_exc_num})", exc):
                 if isinstance(exc, MultiException):
+                    kw['exc_num_prefix'] = f"{full_exc_num}."
                     buff.extend(exc._get_buffer(**kw))
+                    kw['exc_num_prefix'] = exc_num_prefix
```

Comment:
it's "impolite" to modify `kw`:
```suggestion
                    buff.extend(exc._get_buffer(**dict(kw, exc_num_prefix=f"{full_exc_num}."))
```